src/Shared/src/CommandLineUtils/CommandLine/CommandLineApplication.cs

```diff
@@ -161,6 +162,25 @@ public int Execute(params string[] args)
                         var longOptionName = longOption[0];
                         option = command.GetOptions().SingleOrDefault(opt => string.Equals(opt.LongName, longOptionName, StringComparison.Ordinal));
 
+                        if (option == null)
```

Comment:
I missed that this was in shared code. I think we should make this conditional on a boolean being passed in to the `CommandLineApplication`, something like `treatUnmatchedOptionsAsArguments`.